test/baselines/generated-reference/max-clause-depth.html

```diff
@@ -0,0 +1,39 @@
+<!doctype html>
+<head><meta name="viewport" content="width=device-width, initial-scale=1"><meta charset="utf-8">
+<link rel="icon" href="img/favicon.ico">
+</head>
+
+<body><div id="shortcuts-help">
+<ul>
+  <li><span>Toggle shortcuts help</span><code>?</code></li>
+  <li><span>Toggle "can call user code" annotations</span><code>u</code></li>
+
+  <li><span>Jump to search box</span><code>/</code></li>
+  <li><span>Toggle pinning of the current clause</span><code>p</code></li>
+  <li><span>Jump to the <i>n</i><sup>th</sup> pin</span><code>1-9</code></li>
+  <li><span>Jump to the 10<sup>th</sup> pin</span><code>0</code></li>
+  <li><span>Jump to the most recent link target</span><code>`</code></li>
+</ul></div><div id="menu-toggle"><svg xmlns="http://www.w3.org/2000/svg" style="width:100%; height:100%; stroke:currentColor" viewBox="0 0 120 120" width="54" height="54">
+      <title>Menu</title>
+      <path stroke-width="10" stroke-linecap="round" d="M30,60 h60  M30,30 m0,5 h60  M30,90 m0,-5 h60"></path>
+    </svg></div><div id="menu-spacer" class="menu-spacer"></div><div id="menu"><div id="menu-search"><input type="text" id="menu-search-box" placeholder="Search..."><div id="menu-search-results" class="inactive"></div></div><div id="menu-pins"><div class="menu-pane-header">Pins<button class="unpin-all">clear</button></div><ul id="menu-pins-list"></ul></div><div class="menu-pane-header">Table of Contents</div><div id="menu-toc"><ol class="toc"><li><span class="item-toggle">+</span><a href="#one" title="One"><span class="secnum">1</span> One</a><ol class="toc"><li><span class="item-toggle">+</span><a href="#two" title="Two"><span class="secnum">1.1</span> Two</a><ol class="toc"><li><span class="item-toggle-none"></span><a href="#three" title="Three"><span class="secnum">1.2</span> Three</a></li><li><span class="item-toggle-none"></span><a href="#three-again" title="Three Again"><span class="secnum">1.3</span> Three Again</a></li></ol></li><li><span class="item-toggle-none"></span><a href="#two-again" title="Two Again"><span class="secnum">1.4</span> Two Again</a></li></ol></li><li><span class="item-toggle-none"></span><a href="#one-again" title="One Again"><span class="secnum">2</span> One Again</a></li></ol></div></div><div id="spec-container">
+
+<emu-clause id="one">
+  <h1><span class="secnum">1</span> One</h1>
+  <emu-clause id="two">
+    <h1><span class="secnum">1.1</span> Two</h1>
+    <emu-clause id="three">
+      <h1><span class="secnum">1.2</span> Three</h1>
```

Comment:
I would put a depth even deeper than 3 to make sure the flattening happens recursively.